src/czar/CzarRegistry.cc

```diff
@@ -80,13 +80,15 @@ void CzarRegistry::_registryUpdateLoop() {
     string const url = "http://" + _czarConfig->replicationRegistryHost() + ":" +
                        to_string(_czarConfig->replicationRegistryPort()) + "/czar";
     vector<string> const headers = {"Content-Type: application/json"};
+    LOGS(_log, LOG_LVL_INFO, "CzarRegistry::_registryUpdateLoop() getting FQDN");
+    string const fqdn = util::getCurrentHostFqdnBlocking();
```

Comment:
Why would you put this into a separate variable instead of feeding the value directly into:
```
     json const request = json::object(...,
        {"management-host-name", util::getCurrentHostFqdnBlocking()},
        ...
     );
```

Reply:
That's in a loop that will get called a lot, and the function requires contacting a separate program which could block for a significant. By moving it, it's called when the program starts and never again.